apps/extension/src/routes/popup/home/index.tsx

```diff
@@ -33,32 +33,36 @@ export const popupIndexLoader = async (): Promise<Response | PopupLoaderData> =>
 };
 
 export const PopupIndex = () => {
-  const account = useStore(selectedAccount);
-  const { next, previous, setIndex } = useStore(accountsSelector);
+  const {
+    ephemeral,
+    selectedAccount,
+    index,
+    next,
+    previous,
+    setIndex,
+    setEphemeral,
+    setSelectedAccount,
+  } = useStore(accountsSelector);
+
+  useEffect(() => {
+    setSelectedAccount();
+  }, [ephemeral, index, setSelectedAccount]);
```

Comment:
We should not require a useEffect. Please move this logic into the accounts slice.

Reply:
In fact, this is pointing to the fact that `selectedAccount` is really "derived" data and should be a selector.